docs/linting/troubleshooting/Performance.mdx

```diff
@@ -85,6 +85,8 @@ We recommend you do not use the following rules, as TypeScript provides the same
 - `import/namespace`
 - `import/default`
 - `import/no-named-as-default-member`
+- `import/no-unresolved`
+- `import/unambiguous`
```

Comment:
[Question] I don't think TypeScript always enforces this? There's no error reported locally when I write a new `.ts` file that doesn't `import` or `export`.

Unused code detection (e.g. [knip](https://knip.dev)) can, but that's separate.

Reply:
> I don't think TypeScript always enforces this? There's no error reported locally when I write a new .ts file that doesn't import or export.

i am not an expert but i believe that `the [unambiguous](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/unambiguous.md) rule is unnecessary when creating a typescript program.
maybe someone else on the team can weigh in?

Reply:
No - it is actually a necessity!

Without any import/exports TS does treat a file as a script - and in doing that the rules for global augmentation and such change. So including an empty export is still necessary in TS :)

Reply:
hmm, ok.
see the other discussion.
if you accept that the other section should be listed as 
```
- `import/no-unresolved` (as long as you are using `import` over `require`)
```
then maybe this section should be:
```
- `import/unambiguous` (as long as you are using a dead code finder such as `knip` or `ts-prune`)
```
?

Reply:
Another use case for ambiguous scripts would be one that intentionally wants to keep the scripts as is.

Looking at the source for `import/unamiguous`, it's not a performance-critical rule: 
* Its `create`'s logic is mostly a call to an `isModule` method: https://github.com/import-js/eslint-plugin-import/blob/7a21f7e10f18c04473faadca94928af6b8e28009/src/rules/unambiguous.js#L28
* That `isModule` is a loop through nodes: https://github.com/import-js/eslint-plugin-import/blob/7a21f7e10f18c04473faadca94928af6b8e28009/utils/unambiguous.js#L29

Between TypeScript _not performing this check during standard linting_, there being legitimate use cases for code that violates the rule, and the rule being very cheap to run, there's not a need to mention it here.